src/gluonts/dataset/parallelized_loader.py

```diff
@@ -474,17 +474,24 @@ def __iter__(self) -> Iterator[DataBatch]:
 
     def __del__(self):
         # Explicitly load the content from shared memory to delete it
-        # Unfortunately it seems the way the data is pickled prevents efficient implicit GarbageCollection
+        # Unfortunately it seems the way the data is pickled prevents
+        # efficient implicit garbage collection
         try:
             for k in list(self._data_buffer.keys()):
-                res = pickle.loads(self._data_buffer.pop(k).get(self._timeout))
-                del res
+                handle = self._data_buffer.pop(k)
+                # In some cases the workers might have been terminated
+                # before they finished their respective batch,
+                # in which case the results will never be available
+                if handle.ready():
```

Comment:
It seems that this is not the correct way to fix this problem. The ```handle.ready()``` here should always be False since if we need garbage collection to retrieve, these results are always not ready since we never used them. The real problem of the hang bug here should be that the module needed for ```pickle.loads```, for example, numpy and pandas, are deleted in the system before they are used. See the explanation [here](url: https://stackoverflow.com/questions/14986568/python-how-to-ensure-that-del-method-of-an-object-is-called-before-the-mod/14986683#14986683)